cloudformation/external-slurmdbd/external_slurmdbd/external_slurmdbd_stack.py

```diff
@@ -0,0 +1,274 @@
+        )
+        self.vpc = ec2.Vpc.from_lookup(self, "VPC", vpc_id=self.vpc_id.value_as_string)
+
+        self.subnet_id = CfnParameter(
+            self, "SubnetId", type="String", description="The Subnet to be used for the Slurmdbd stack."
+        )
+        self.subnet = ec2.Subnet.from_subnet_id(self, "subnet", subnet_id=self.subnet_id.value_as_string)
+
+        # define Target Group
+        self._external_slurmdbd_target_group = self._add_external_slurmdbd_target_group()
+
+        # define Network Load Balancer (NLB)
+        self._external_slurmdbd_nlb = self._add_external_slurmdbd_load_balancer(
+            target_group=self._external_slurmdbd_target_group
+        )
+
+        # Define additional CloudFormation parameters for dna.json to pass to cookbook
+        self.dbms_uri = CfnParameter(self, "DBMSUri", type="String", description="DBMS URI for Slurmdbd.")
+        self.dbms_username = CfnParameter(
+            self, "DBMSUsername", type="String", description="DBMS Username for Slurmdbd."
+        )
+        self.dbms_password_secret_arn = CfnParameter(
+            self, "DBMSPasswordSecretArn", type="String", description="Secret ARN for DBMS password."
+        )
+        self.dbms_database_name_para = CfnParameter(
+            self, "DBMSDatabaseName", type="String", description="DBMS Database Name for Slurmdbd."
+        )
+        self.munge_key_secret_arn = CfnParameter(
+            self, "MungeKeySecretArn", type="String", description="Secret ARN for Munge key."
+        )
+
+        # use cfn-init and cfn-hup configure instance
+        self._cfn_init_config = self._add_cfn_init_config()
+
+        # create management security group with SSH access from anywhere (TEMPORARY!)
+        self._ssh_server_sg, self._ssh_client_sg = self._add_management_security_groups()
+
+        # create Launch Template
+        self._launch_template = self._add_external_slurmdbd_launch_template()
+
+        # define EC2 Auto Scaling Group (ASG)
+        self._external_slurmdbd_asg = self._add_external_slurmdbd_auto_scaling_group()
+
+        # Create a CloudWatch log group
+        self._log_group = self._add_cloudwatch_log_group()
+
+        # define IAM role and necessary IAM policies
+        self._role = self._add_iam_role()
+
+    def _add_cfn_init_config(self):
+        dna_json_content = {
+            "dbms_uri": self.dbms_uri_param.value_as_string,
+            "dbms_username": self.dbms_username_param.value_as_string,
+            "dbms_database_name": self.dbms_database_name_para.value_as_string,
+            "dbms_password_secret_arn": self.dbms_password_secret_arn_param.value_as_string,
+            "munge_key_secret_arn": self.munge_key_secret_arn_param.value_as_string,
+            "region": self.stack.region,
+            "stack_name": self.stack.stack_name,
+            "nlb_dns_name": self._external_slurmdbd_nlb.load_balancer_dns_name,
+            "if_external_slurmdbd": True,
```

Comment:
Rename this parameter to `is_external_slurmdbd`.

Reply:
Done!